fpm/test/new_test/new_test.f90

```diff
@@ -0,0 +1,286 @@
+         case('G')
+          expected=[ character(len=80)::&
+          &'G/app','G/fpm.toml','G/README.md','G/test','G/app/main.f90','G/test/main.f90']
+         case('BB')
+          expected=[ character(len=80)::&
+          &'BB/fpm.toml','BB/README.md','BB/src','BB/test','BB/src/BB.f90','BB/test/main.f90']
+         case('CC')
+          expected=[ character(len=80)::&
+          &'CC/app','CC/fpm.toml','CC/README.md','CC/src','CC/test','CC/app/main.f90','CC/src/CC.f90','CC/test/main.f90']
+         case default
+            write(*,*)'ERROR: internal error. unknown directory name ',trim(directories(i))
+            stop 4
+         end select
+         !! MSwindows has hidden files in it
+         call list_files(trim(directories(i)), file_names,recurse=.true.)
+         if(allocated(fnames))deallocate(fnames)
+         allocate(character(len=0) :: fnames(0))
+         do j=1,size(file_names)
+           if(file_names(j)%s(1:1).eq.'.'.or.index(file_names(j)%s,'/.').ne.0.or.index(file_names(j)%s,'\.').ne.0)cycle
+           fnames=[character(len=max(len(fnames),len(file_names(j)%s))) :: fnames,file_names(j)%s]
+         enddo
+         write(*,'(*(g0))',advance='no')'>>>DIRECTORY ',trim(directories(i)),':   '
+         write(*,'(*(g0:,", "))')( file_names(j)%s, j=1,size(file_names) )
+         if(size(expected).ne.size(fnames))then
+            write(*,*)'unexpected number of files in file list=',size(fnames),' expected ',size(expected)
+            tally=[tally,.false.]
+            cycle TESTS
+         else
+            select case (get_os_type()) 
+            case (OS_UNKNOWN, OS_LINUX, OS_MACOS, OS_CYGWIN, OS_SOLARIS, OS_FREEBSD)
+            case (OS_WINDOWS) 
+               do j=1,size(expected)
+                  expected(j)=u2d(expected(j))
+               enddo 
+            case default
+               write(*,*)'ERROR: unknown OS. Stopping test'
+               stop 3
+            end select 
+            do j=1,size(expected)
+               if( .not.any(fnames(j)==expected) )then
+                  tally=[tally,.false.]
+                  write(*,'("ERROR: EXPECTED ",*(g0:,", "))')( trim(expected(k)), k=1,size(expected) )
+                  write(*,'(*(g0))')'       NO MATCH FOR ',fnames(j)
+                  cycle TESTS
+               endif
+            enddo
+            tally=[tally,.true.]
+         endif
+      endif
+   enddo TESTS
+   write(*,'("TALLY=",*(g0))')tally
+   if(all(tally))then
+      write(*,'(*(g0))')'PASSED: all ',count(tally),' tests passed '
+   else
+      write(*,*)'FAILED: PASSED=',count(tally),' FAILED=',count(.not.tally)
+      stop 5
+   endif
+!-----------------------------------------------------------------------------------------------------------------------------------
+contains
+!-----------------------------------------------------------------------------------------------------------------------------------
+function u2d(pathname) result(dos)
```

Comment:
See `windows_path` in `fpm_filesystem`: <https://github.com/fortran-lang/fpm/blob/49de89e8608da15a04bed388e2d1774e1737acd8/fpm/src/fpm_filesystem.f90#L342>